### PR TITLE
[TF] Add Implementation of rnn-concise

### PR DESCRIPTION
**Pull request by floriandonhauser**
This pull requests adds the concise implementation for RNN for TF.
It is heavily based on @terrytangyuan previous suggestion (PR #1467) and I want to thank him for his work.
Some small changes for the previous chapter for rnn-scratch were necessary and I tested these changes.
A screenshot of the result of rnn-concise is attached:
rnn-concise-tf

I created a new PR since I had no write access to the repo without my own fork.
As soon as this PR is merged I will continue with the TF version for the following chapters on GRU (9.1) and LSTM (9.2).
Looking forward to your feedback :)

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1677/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1677/

**Comment by floriandonhauser** [comment]
For the final version, someone of course still needs to add the Discussions tab :)

**Review by terrytangyuan — APPROVED** [review]
This is really good start. Thanks!

File: chapter_recurrent-neural-networks/rnn-concise.md [hunk 1]
```diff
@@ -196,6 +226,24 @@ class RNNModel(nn.Module):
...
+```{.python .input}
+#@tab tensorflow
+#@save
+class RNNModel(tf.keras.layers.Layer):
+    def __init__(self, rnn_layer, vocab_size, **kwargs):
+        super(RNNModel, self).__init__(**kwargs)
+        self.rnn = rnn_layer
+        self.vocab_size = vocab_size
+        self.dense = tf.keras.layers.Dense(vocab_size)
+    def call(self, inputs, state, _):
```
Comment (terrytangyuan) [inline chapter_recurrent-neural-networks/rnn-concise.md:238]: Need new lines between different functions (can be addressed later with other PRs).

**Comment by astonzhang** [comment]
@floriandonhauser  Thanks! The discussion link is added https://github.com/d2l-ai/d2l-en/commit/f5eb1d6268a79f30be4e0c57cf31f242c60acc11

**Pull request merged**